hail/python/hailtop/hailctl/auth/create_user.py

```diff
@@ -0,0 +1,46 @@
+import asyncio
+
+from hailtop.utils import sleep_and_backoff
+from hailtop.auth import async_create_user, async_get_user
+
+
+def init_parser(parser):
+    parser.add_argument("username", type=str,
+                        help="User name to create.")
+    parser.add_argument("login_id", type=str,
+                        help="Login ID to be used with OAuth. This is the object ID in Azure and the email address in GCP.")
+    parser.add_argument("--developer", default=False, action='store_true',
+                        help="User should be a developer.")
+    parser.add_argument("--service-account", default=False, action='store_true',
+                        help="User should be a service account.")
+    parser.add_argument("--namespace", "-n", type=str,
+                        help="Specify namespace for auth server.  (default: from deploy configuration)")
+    parser.add_argument("--no-wait", default=False, action='store_true',
+                        help="Do not wait for the creation of the user to finish")
+
+
+async def async_main(args):
+    try:
+        await async_create_user(args.username, args.login_id, args.developer, args.service_account, args.namespace)
+
+        if args.no_wait:
+            return
+
+        async def _poll():
+            delay = 5
+            while True:
+                user = await async_get_user(args.username, args.namespace)
+                if user['state'] == 'active':
+                    print(f"Created user '{args.username}'")
+                    return
+                assert user['state'] == 'creating'
+                delay = await sleep_and_backoff(delay)
+
+        await asyncio.wait_for(_poll(), 300)
```

Comment:
I'm open to arguments in favor, but I don't think we should have this timeout. The HTTP requests themselves already have time outs. This only triggers if it takes longer than 5 minutes to create a user account. I'm not sure there's a compelling reason for that. I would expect a CLI tool to just wait until it completes. I can always ctrl-c if I want to abort. If this is used inside a test or w/e, that test will have a timeout.